src/Symfony/Component/HttpKernel/EventListener/HttpExceptionListener.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\EventListener;
+
+use Symfony\Component\Debug\Exception\FlattenException;
+use Symfony\Component\Debug\ExceptionHandler;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Event\GetResponseForExceptionEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * HttpExceptionListener.
+ *
+ * @author Chris McCafferty <cilefen@gmail.com>
+ */
+class HttpExceptionListener implements EventSubscriberInterface
+{
+    public static function getSubscribedEvents()
+    {
+        return array(
+          KernelEvents::EXCEPTION => array(
+            array('onKernelException', -256),
+          ),
+        );
+    }
+
+    public function onKernelException(GetResponseForExceptionEvent $event)
+    {
+        if ($event->hasResponse()) {
+            return;
+        }
+
+        $flattenException = FlattenException::create($event->getException());
+        $handler = new ExceptionHandler(false);
+        $response = new Response($handler->getContent($flattenException));
+        $event->setResponse($response);
```

Comment:
@nicolas-grekas This does not actually work as expected in the dev environment. In fact I do no understand how to detect the environment without injecting it.

Reply:
@nicolas-grekas ... and indeed I may be way "off the mark" in terms of what you wrote about delegating to the ExceptionHandler.

Reply:
Actually, I was wrong, `kernel.debug` may be required to populate the first argument of ExceptionHandler's constructor.
About the implementation, I suggest to borrow from `ExceptionHandler::sendPhpResponse()`

Reply:
@nicolas-grekas hmm but should this new listener really stream/echo the response to the client directly? I think it should just set the response on the event and nothing else?

Reply:
We should not call sendPhpResponse but take inspiration from it. Eg header management is missing right now .